examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,585 @@
+  static constexpr auto description()
+  {
+    return "Parametric Equalizer, using Butterworth equations for all filters except "
+           "the bandshelf filter (which is computed with Chebyshev Type I equations).";
+  }
+  static constexpr auto uuid() { return "ec972d12-edba-4a9c-9736-9176efad17e3"; }
+
+  static constexpr int bandsNb = 8;  // EQ Bands number
+  static constexpr int maxOrder = 7; //Filters maximum order
+  struct ins
+  {
+    struct
+    {
+      static constexpr auto name() { return "Input"; }
+      const double** samples;
+      int channels;
+      float sampleRate;
+    } audio;
+
+    EQBand band1, band2, band3, band4, band5, band6, band7, band8;
+
+    // Common filters properties
+    filterToggle toggle1, toggle2, toggle3, toggle4, toggle5, toggle6, toggle7, toggle8;
+    order<maxOrder> order1, order2, order3, order4, order5, order6, order7, order8;
+
+    // Specific filters properties (not all filters require them)
+    cutoffFrequency cutoffFreq1, cutoffFreq2, cutoffFreq3, cutoffFreq4, cutoffFreq5,
+        cutoffFreq6, cutoffFreq7, cutoffFreq8;
+    centerFrequency centerFreq1, centerFreq2, centerFreq3, centerFreq4, centerFreq5,
+        centerFreq6, centerFreq7, centerFreq8;
+    gain gain1, gain2, gain3, gain4, gain5, gain6, gain7, gain8;
+    frequencyBandWidth freqBandWidth1, freqBandWidth2, freqBandWidth3, freqBandWidth4,
+        freqBandWidth5, freqBandWidth6, freqBandWidth7, freqBandWidth8;
+    passbandRipple passbandRipple1, passbandRipple2, passbandRipple3, passbandRipple4,
+        passbandRipple5, passbandRipple6, passbandRipple7, passbandRipple8;
+
+  } inputs;
+
+  //                                      Available filters
+  static constexpr auto chans = 2;
+  Dsp::Filter * lowpassFilter, * highpassFilter, * bandpassFilter, * bandstopFilter, * lowshelfFilter, * highshelfFilter, * bandshelfFilter;
+
+  // EQ Bode diagram calculator
+  static constexpr auto plotPoints = 5000;
+  EQDiagramCalculator<plotPoints, bandsNb> diagramCalculator;
+  bool  toRecompute;
+  struct processor_to_ui
+  {
+    const float* xInput;
+    const float* yInput;
+  };
+  std::function<void(processor_to_ui)> send_message;
+
+  struct
+  {
+    struct
+    {
+      static constexpr auto name() { return "Output"; }
+      double** samples;
+      int channels;
+    } audio;
+  } outputs;
+  struct setup
+  {
+    float rate{};
+  };
+
+  void prepare(setup s)
+  {
+    std::cout << "preparing" << std::endl;
+    inputs.audio.sampleRate = s.rate;
+    diagramCalculator.sampleRate = s.rate;
+    inputs.band1.id = 0;
+    inputs.band2.id = 1;
+    inputs.band3.id = 2;
+    inputs.band4.id = 3;
+    inputs.band5.id = 4;
+    inputs.band6.id = 5;
+    inputs.band7.id = 6;
+    inputs.band8.id = 7;
+    std::cout << "initializing filters with pointers" << std::endl;
+    lowpassFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::LowPass<maxOrder>, chans>; // crashes when cutoffFreq = 0 (should never happen)
```

Comment:
i'm curious, why not initialize them as values directly instead of new'ing them ?

Reply:
I had in mind to maybe allow to choose the filter type (Butterworth, Chebyshev, etc) in the future. But now that I think about it, I would still need to have an instance of each filter type (it'd be better for efficiency)...

Reply:
yep, preallocating filter objects would be the way to go to ensure no cracks